pkg/config/config_test.go

```diff
@@ -88,5 +91,308 @@ func TestLoadFile(t *testing.T) {
+	receiverConfig := newReceiverTestConfig([]string{"Name"}, []string{})
+
+	var config testConfig
+
+	// No receivers
+	config = testConfig{Defaults: defaultsConfig, Receivers: []*receiverTestConfig{}, Template: "jiralert.tmpl"}
+	configErrorTestRunner(t, config, "no receivers defined")
+
+	// No template
+	config = testConfig{Defaults: defaultsConfig, Receivers: []*receiverTestConfig{receiverConfig}}
+	configErrorTestRunner(t, config, "missing template file")
+}
+
+// Tests regarding mandatory keys
+// No tests for auth keys here. They will be handled separately
+func TestRequiredReceiverConfigKeys(t *testing.T) {
+	type testCase struct {
+		missingField string
+		errorMessage string
+	}
+
+	// We'll remove one key at a time and check that the error is the expected one
+	testTable := []testCase{
+		{"Name", "missing name for receiver"},
+		{"APIURL", `missing api_url in receiver "Name"`},
+		{"Project", `missing project in receiver "Name"`},
+		{"IssueType", `missing issue_type in receiver "Name"`},
+		{"Summary", `missing summary in receiver "Name"`},
+		{"ReopenState", `missing reopen_state in receiver "Name"`},
+		{"ReopenDuration", `missing reopen_duration in receiver "Name"`},
+	}
+
+	mandatory := mandatoryReceiverFields()
+	for _, test := range testTable {
+		fields := removeFromStrSlice(mandatory, test.missingField)
+
+		// non-empty defaults as we don't handle the empty defaults case yet
+		defaultsConfig := newReceiverTestConfig([]string{}, []string{"Priority"})
+		receiverConfig := newReceiverTestConfig(fields, []string{})
+		config := testConfig{
+			Defaults:  defaultsConfig,
+			Receivers: []*receiverTestConfig{receiverConfig},
+			Template:  "jiratemplate.tmpl",
+		}
+		configErrorTestRunner(t, config, test.errorMessage)
+	}
+
+}
+
+// Auth keys error scenarios
+func TestAuthKeysErrors(t *testing.T) {
+	type testCase struct {
+		receiverTestConfigMandatoryFields []string
+		errorMessage                      string
+	}
+
+	mandatory := mandatoryReceiverFields()
+
+	// Test cases
+	// * missing user
+	// * missing password
+	// * specifying user/password and PAT auth
+	testTable := []testCase{
+		{
+			removeFromStrSlice(mandatory, "User"),
+			`missing authentication in receiver "Name"`,
+		},
+		{
+			removeFromStrSlice(mandatory, "Password"),
+			`missing authentication in receiver "Name"`,
+		},
+		{
+			append(mandatory, "PersonalAccessToken"),
+			"bad auth config in defaults section: user/password and PAT authentication are mutually exclusive",
+		},
+	}
+
+	minimalReceiverTestConfig := newReceiverTestConfig([]string{"Name"}, []string{})
+	for _, test := range testTable {
+		defaultsConfig := newReceiverTestConfig(test.receiverTestConfigMandatoryFields, []string{})
+		config := testConfig{
+			Defaults:  defaultsConfig,
+			Receivers: []*receiverTestConfig{minimalReceiverTestConfig},
+			Template:  "jiralert.tmpl",
+		}
+
+		configErrorTestRunner(t, config, test.errorMessage)
+	}
+}
```

Comment:
This is great! 

But there is one pattern we could reuse here to simplify code (and have consistency).

```suggestion
	mandatory := mandatoryReceiverFields()

	minimalReceiverTestConfig := newReceiverTestConfig([]string{"Name"}, []string{})
	for _, test := range []struct {
		receiverTestConfigMandatoryFields []string
		errorMessage                      string
	}{
	      {
	      	      //  Missing user.
			removeFromStrSlice(mandatory, "User"),
			`missing authentication in receiver "Name"`,
		},
		{
		        // Missing password.
			removeFromStrSlice(mandatory, "Password"),
			`missing authentication in receiver "Name"`,
		},
		{
		       // Specifying user/password and PAT auth.
			append(mandatory, "PersonalAccessToken"),
			"bad auth config in defaults section: user/password and PAT authentication are mutually exclusive",
		},
	} {
		defaultsConfig := newReceiverTestConfig(test.receiverTestConfigMandatoryFields, []string{})
		config := testConfig{
			Defaults:  defaultsConfig,
			Receivers: []*receiverTestConfig{minimalReceiverTestConfig},
			Template:  "jiralert.tmpl",
		}

		configErrorTestRunner(t, config, test.errorMessage)
	}
}
```

Reply:
Fixed this everywhere in b2861cf